### ISSUE TITLE
[Feature] Notifications when printer sends action commands

### ISSUE THREAD

**Issue by Doormat1** [TheSpaghettiDetective/TheSpaghettiDetective#392]
I have recently installed the smart filament sensor on my ender 3 pro runnin an skr mini e3 v2. 
I have host commands enabled in my firmware so when the printer sends the m600 command when the sensor detects an issue octoprint pauses. 
Is there anyway TSD could notify me when these happen as well?

**Comment by kennethjiang** [comment]
This is a really good feature request.

Does OctoPrint pick up these m600 command, and hopefully even make it available with its APIs? If so, it'll be a lot easier to implement. Will appreciate it if you can post a few more pointers about how widely accepted m600 command is, and the status of OctoPrint support of that.

**Comment by Doormat1** [comment]
Yes Octoprint picks them up and pauses the print with a notification on the
screen until the user deals with it.

**Comment by kennethjiang** [comment]
That's a good news!

Can you dig up a bit more info, such as if OctoPrint will send an event to all plugins on m600? I don't have a printer that I can test with right now.

**Comment by Doormat1** [comment]
This is the octoprint documentation
https://docs.octoprint.org/en/master/features/action_commands.html
https://docs.octoprint.org/en/master/bundledplugins/action_command_prompt.html

hope they help

**Comment by kennethjiang** [comment]
This is awesome! Should make things a lot easier! Thank you @Doormat1

**Comment by Doormat1** [comment]
Glad to help! People need to enable it in config adv, I got the
instructions from the teaching tech video on the smart filament sensor

**Comment by kennethjiang** [comment]
Gotcha. Thanks again for the tips so that I don't have to waste time searching for info. :)

**Comment by dfredell** [comment]
My use of m600 is slightly different. I sometimes do a two colored print on my Ender 3 and switch the filliment part way through the print. The M600 is added to the gcode via Cura at layer X. 
Currently I have Octoslack installed to notify me of M600s.

**Comment by kennethjiang** [comment]
This is a really good info @dfredell . Will definitely give us a jump start!
